WhatsLab/src/App.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react'
+import styled from 'styled-components';
+
+
+const RedSpan = styled.span `
+color: red;
+`
+
+class App extends React.Component {
+    state = {
+        usuario: [''],
+        mensagem: [''],
+        valorInputUsuario: [''],
+        valorInputMensagem: ['']
+
+    }
+
+
+onClickEnviar = () => {
+    const novoUsuario = this.state.valorInputUsuario;
+    const novosUsuarios = [novoUsuario, ...this.state.usuario];
+    this.setState ({usuario: novosUsuarios})
+
+    const novaMensagem = this.state.valorInputMensagem;
+    const novasMensagens = [novaMensagem, ...this.state.mensagem];
+    this.setState ({mensagem: novasMensagens})    
```

Comment:
Vejam o comentário acima, que eu falei sobre o estado inicial. Com a mudança que eu disse pra vcs: 

```
state = {
  listaDeMensagens: [
      usuario: "",
      mensagem: ""
  ]
}
```

Se o estado de vcs estivesse assim, bastaria vcs usarem um `this.setState()` para mudar o estado, e não um para a propriedade usuario e outro para a propriedade mensagem. De novo, peço para que vcs deem uma olhada no vídeo do Darvas, de manipulação de Arrays em React, que lá ele deixa essa criação de estados iniciais com array bem clara.